src/Microsoft.OData.Edm/Vocabularies/Annotations/IEdmTermExtensions.cs

```diff
@@ -0,0 +1,300 @@
+                    {
+                        return new EdmFloatingConstant(dbl.Value);
+                    }
+
+                    throw new FormatException(Strings.ValueParser_InvalidFloatingPoint(value));
+
+                case EdmPrimitiveTypeKind.Guid:
+                    if (EdmValueParser.TryParseGuid(value, out Guid? gd))
+                    {
+                        return new EdmGuidConstant(gd.Value);
+                    }
+
+                    throw new FormatException(Strings.ValueParser_InvalidGuid(value));
+
+                case EdmPrimitiveTypeKind.Int16:
+                case EdmPrimitiveTypeKind.Int32:
+                    if (EdmValueParser.TryParseInt(value, out int? intNum))
+                    {
+                        return new EdmIntegerConstant(intNum.Value);
+                    }
+
+                    throw new FormatException(Strings.ValueParser_InvalidInteger(value));
+
+                case EdmPrimitiveTypeKind.Int64:
+                    if (EdmValueParser.TryParseLong(value, out long? longNum))
+                    {
+                        return new EdmIntegerConstant(longNum.Value);
+                    }
+
+                    throw new FormatException(Strings.ValueParser_InvalidLong(value));
+
+                case EdmPrimitiveTypeKind.String:
+                    return new EdmStringConstant(value);
+
+                case EdmPrimitiveTypeKind.TimeOfDay:
+                    if (EdmValueParser.TryParseTimeOfDay(value, out TimeOfDay? tod))
+                    {
+                        return new EdmTimeOfDayConstant(tod.Value);
+                    }
+
+                    throw new FormatException(Strings.ValueParser_InvalidTimeOfDay(value));
+            }
+
+            throw new NotSupportedException(Strings.EdmVocabularyAnnotations_TermTypeNotSupported(typeReference.FullName()));
+        }
+
+        /// <summary>
+        /// Returns an IEdmExpression for an EDM path type.
+        /// </summary>
+        /// <param name="pathType">Reference to the type of term.</param>
+        /// <param name="value">The value to parse.</param>
+        /// <returns>
+        /// Expression for the value.
+        /// </returns>
+        private static IEdmExpression BuildEdmPathExp(IEdmPathType pathType, string value)
+        {
+            switch (pathType.PathKind)
+            {
+                case EdmPathTypeKind.AnnotationPath:
+                    return new EdmAnnotationPathExpression(value);
+
+                case EdmPathTypeKind.PropertyPath:
+                    return new EdmPropertyPathExpression(value);
+
+                case EdmPathTypeKind.NavigationPropertyPath:
+                    return new EdmNavigationPropertyPathExpression(value);
+
+                default:
+                    return new EdmPathExpression(value);
+            }
+        }
+
+        public static bool IsEqual(this IEdmExpression left, IEdmExpression right)

```

Comment:
I'm not seeing that this is actually called anywhere. Also, an `IEqualityComparer` implementation might be better